src/app/components/list/list.js

```diff
@@ -122,8 +121,6 @@ listControllers.controller('ListCtrl', ['$scope', '$routeParams', '$location', '
       });
     };
     $scope.list.$promise.then(listCallback);
-    $scope.list.$httpPromise.then(listCallback);
```

Comment:
Ok so good catch on the fact that it makes the user request twice, however it is still needed and the issue needs to be fixed in another way. $promise.then is called only when the list has already been cached (in which case, with your removal, things would work). When the list hasn't already been cached however, $promise.then is NOT called (but $httpPromise.then is), which means that with your removal, when the list has not been cached already, listCallback would not be called.